internal/venice-common/src/main/java/com/linkedin/venice/writer/update/UpdateBuilderImpl.java

```diff
@@ -131,6 +133,10 @@ private Exception validateUpdateRecordIsSerializable(GenericRecord updateRecord)
     try {
       serializer.serialize(updateRecord);
     } catch (Exception serializationException) {
+      if (serializationException instanceof AvroRuntimeException || serializationException instanceof UnresolvedUnionException) {
+        Object unresolvedDatum = serializationException.getUnresolvedDatum();
```

Comment:
That is probably the compilation failure... `UnresolvedUnionException` contains the `getUnresolvedDatum()` API, but `Exception` does not. Since `serializationException` is of type `Exception`, the compilation fails.

I think we should specifically `catch (UnresolvedUnionException e)` rather than catching `Exception`, and then we could use `getUnresolvedDatum` on it directly.

Reply:
I made some changes to just catch UnresolvedUnionException specifically, let me know if this works